doc/adr/0009-use-crds-in-declarative-interface.md

```diff
@@ -0,0 +1,74 @@
+# 9. Use CRDs in declarative interface
+
+Date: 2024-02-26
+
+## Status
+
+Proposed
+
+## Context
+
+The motivation for this decision is a desire to provide a better, more
+intuitive interface between the controller and the user. Custom
+Resource Definitions are now the standard way of providing extended
+interfaces in Kubernetes. They are as good as any other format for
+podman based sites.
+
+In addition to allowing the user to configure skupper through more
+clearly defined types, CRDs also give us a familiar way to get
+information back from the controller to the user through the status
+sections.
+
+Only the controller image should be tied to using the CRDs
+directly. The config-sync image should not need to interact with
+them. That ensures that should other aprpoaches be necessary we
+minimise the amount of work required.
+
+## Decision
+
+Instead of creating ConfigMaps and Secrets with annotations users will
+create custom resources to configure Skupper.
+
+## Consequences
+
+Standard RBAC can be used to limit, at a coarse granularity, what
+aspects of Skupper users of a cluster are free to configure.
+
+For example, if a user is denied permission to create Site resources,
+then they will be unable to initialise skupper for a particular
+namespace. If they are denied the permission to create Link resources,
+they will not be able to link a Site to any other. If they are denied
+the permission to create Connector resources, they will not be able to
+expose workloads from that namespace. If they are denied permission to
+create Listener resources they will not be able to offer
+Skupper-backed services to that namespace.
+
+The use of CRDs also allows basic rules about what is allowed to be
+enforced in a way that is not easy with annotations and labels. As an
+example we can define the connector resource to require values for
+port and routingKey and to only accept one of host or selector, not
+both. This makes it possible to avoid certain misconfigurations.
+
+The controller can also use the status section of defined CRDs to
+provide information back to the user, e.g. regarding errors or
+indicating that configuration has been applied. This allows a better
+user experience using just the standard kubectl tool.
```

Comment:
My initial tought would be to have a similar behavior on podman sites, by updating the statuses of CRs.
But in this case users won't simply have the benefit of kubectl, instead they'd still need the CLI for a better
user experience (other than using ls, grep and other shell tools).

Reply:
That sounds reasonable.